src/pool.spec.ts

```diff
@@ -401,6 +401,62 @@ describe("Constant product pool", () => {
     await pool.updateState();
     expect(pool.state.totalLiquidity).toBe(1000);
   });
+
+  it("zap e2e scenario", async () => {
+    const { account, pool } = await makeFreshTestBed({
+      poolType: "CONSTANT_PRODUCT",
+    });
+
+    expect(pool.state).toEqual({
+      totalLiquidity: 0,
+      totalPrimary: 0,
+      totalSecondary: 0,
+      primaryAssetPrice: 0,
+      secondaryAssetPrice: 0,
+    });
+
+    // Opt in for liquidity asset
+    const liqOptInTx = await pool.liquidityAsset.prepareOptInTx(account.addr);
+    await signAndSend(liqOptInTx, account);
+
+    // Add first liquidity
+    const liquidityAddition = pool.prepareAddLiquidity({
+      primaryAssetAmount: 100_000,
+      secondaryAssetAmount: 100_000,
+    });
+    const addLiqTxGroup = await liquidityAddition.prepareTxGroup(account.addr);
+    expect(addLiqTxGroup.transactions.length).toBe(3);
+    await signAndSend(addLiqTxGroup, account);
+    await pool.updateState();
+    expect(pool.state).toEqual({
+      totalLiquidity: 100_000,
+      totalPrimary: 100_000,
+      totalSecondary: 100_000,
+      primaryAssetPrice: 1,
+      secondaryAssetPrice: 1,
+    });
+
+    // Zapping
+    const zapParams = pool.getZapParams({ secondaryAssetAmount: 10_000 });
+    expect(zapParams.swapDeposited + zapParams.secondaryAddLiq).toBe(10_000n);
+    expect(zapParams.swapReceived >= zapParams.primaryAddLiq).toBe(true);
```

Comment:
Isn't it better to check the whole object?

Reply:
Yeah, I wanted to do so, but then I saw that tests above do something like this